internal/controller/pipelinerollout_controller.go

```diff
@@ -166,7 +166,9 @@ func (r *PipelineRolloutReconciler) reconcile(
 		if err != nil {
 			return false, err
 		}
-	} else if rolloutChildOp == RolloutChildUpdate {
+
+		kubernetes.SetAnnotation(pipelineRollout, apiv1.KeyHash, childResouceSpecHash)
+	} else {
```

Comment:
did you see my comment [here](https://github.com/numaproj/numaplane/pull/56#discussion_r1639024255)? I actually think it does make sense for this logic to be when we're still in the process of trying to update...unless I'm missing something?